core/domain/mailchimp_services.py

```diff
@@ -0,0 +1,172 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for API requests to Mailchimp."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+from mailchimp3 import MailChimp
+from mailchimp3.mailchimpclient import MailChimpError
+
+import feconf
+import hashlib
+import python_utils
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+def _get_mailchimp_class():
+    """Returns the mailchimp api class. This is separated into a separate 
+    function to facilitate testing. 
+
+    NOTE: No other functionalities should be added to this function.
+    """
+    if not feconf.MAILCHIMP_API_KEY:
+        raise Exception('Mailchimp API key is not available.')
+
+    if not feconf.MAILCHIMP_USERNAME:
+        raise Exception('Mailchimp username is not set.')
+
+    return MailChimp(
+        mc_api=feconf.MAILCHIMP_API_KEY, mc_user=feconf.MAILCHIMP_USERNAME)
+
+
+def get_subscriber_hash(email):
+    """Returns Mailchimp subscriber hash from email.
+    
+    Args:
+        email: str. The email ID of the user.
```

Comment:
Drop "ID". But if this is being used as an ID, explain it (e.g. if it's used as the ID in Mailchimp, add a sentence explaining that).

Reply:
It is used as the identifier for the user in the mailchimp database. Added explanation in the docstring.